src/postGithubComments.ts

```diff
@@ -38,21 +38,32 @@ for (const path of metadataFilePaths) {
                 somethingChanged = true;
                 break;
             default:
-                infrastructureFailed = true;
+                infrastructureFailures.set(status, (infrastructureFailures.get(status) ?? 0) + 1)
                 break;
         }
     }
 }
 
-let summary: string;
-if (somethingChanged && (isTopReposRun || !infrastructureFailed)) {
```

Comment:
It took me way too long to understand the meaning of this logic, so I've simplified things down to make it clear that the intent was to not report infra failures for top-repos runs.